.github/workflows/workflow-build-and-deploy-scaleway.yml

```diff
@@ -56,8 +60,63 @@ on:
         required: true
 
 jobs:
-  build-and-deploy:
-    name: Build & Deploy to ${{ inputs.environment }}
+  monorepo-check:
+    name: Monorepo or not
+    runs-on: ubuntu-latest
+    outputs:
+      monorepo-check: ${{ steps.file_check.outputs.monorepo-check }}
+      deployments: ${{ steps.changed-paths-filter.outputs.matrix }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: .github
+      - name: Check if file exists
+        id: file_check
+        run: |
+          if [ -f .github/monorepo-rollout-filters.yaml ];
+          then echo "monorepo-check=true" >> $GITHUB_OUTPUT;
+          else echo "monorepo-check=false" >> $GITHUB_OUTPUT; fi
+      - name: Generate Matrix
+        id: changed-paths-filter
+        if: ${{ steps.file_check.outputs.monorepo-check == 'true' }}
+        uses: wisemen-digital/devops-ga-changed-paths-filter@main
+        with:
+          filter-file: .github/monorepo-rollout-filters.yaml
+          changes-override: ${{ inputs.monorepo-deployments}}
+
+  monorepo-build:
+    name: Build to ${{ matrix.deployment.path }}
+    needs: [monorepo-check]
+    if: ${{ needs.monorepo-check.outputs.monorepo-check == 'true' }}
```

Comment:
This can just check if the array is empty. See the `k8s-workflow-rollout-to-scaleway` workflow on how to do this.